server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/VariableDefinitionUtil.java

```diff
@@ -0,0 +1,424 @@
+ */
+@UtilityClass
+@Slf4j
+public class VariableDefinitionUtil {
+  public static final int LEVEL_MNEMONIC = 0;
+  public static final int LEVEL_01 = 1;
+  public static final int LEVEL_66 = 66;
+  public static final int LEVEL_77 = 77;
+  public static final int LEVEL_88 = 88;
+  public static final int AREA_A_FINISH = 10;
+
+  public static final String EMPTY_STRUCTURE_MSG = "semantics.emptyStructure";
+  public static final String TOO_MANY_CLAUSES_MSG = "semantics.tooManyClauses";
+  public static final String PREVIOUS_WITHOUT_PIC_FOR_88 = "semantics.previousWithoutPicFor88";
+  public static final String AREA_A_WARNING = "CobolVisitor.AreaAWarningMsg";
+  public static final String NUMBER_NOT_ALLOWED_AT_TOP = "semantics.numberNotAllowedAtTop";
+  public static final String NO_STRUCTURE_BEFORE_RENAME = "semantics.noStructureBeforeRename";
+  public static final String CHILD_TO_RENAME_NOT_FOUND = "semantics.childToRenameNotFound";
+  public static final String INCORRECT_CHILDREN_ORDER = "semantics.incorrectChildrenOrder";
+  public static final String CANNOT_BE_RENAMED = "semantics.cannotBeRenamed";
+  public static final String GLOBAL_NON_01_LEVEL_MSG = "semantics.globalNon01Level";
+  public static final String GLOBAL_TOO_MANY_DEFINITIONS = "semantics.globalTooManyDefinitions";
+  public static final String REDEFINE_IMMEDIATELY_FOLLOW = "semantics.redefineImmediatelyFollow";
+  public static final String LEVELS_MUST_MATCH = "semantics.levelsMustMatch";
+  public static final String REDEFINED_CONTAIN_VALUE = "semantics.redefinedContainValue";
+
+  public static final ErrorSeverity SEVERITY = ERROR;
+
+  /**
+   * Process nodes with {@link VariableDefinitionNode} as a child and convert {@link VariableDefinitionNode} into
+   * an appropriate {@link VariableNode}.
+   * <p>
+   * The method do the following things: <p>
+   * - delete all VariableDefinitionNode from the node <p>
+   * - convert VariableDefinitionNode into appropriate VariableNode <p>
+   * - build correct tree from VariableNodes and attach it to the node <p>
+   * - register variable definitions into nearest ProgramNode
+   *
+   * @param node the node with VariableDefinitionNodes
+   * @return the list of errors
+   */
+  public List<SyntaxError> processNodeWithVariableDefinitions(Node node) {
+    Deque<VariableDefinitionNode> variableDefinitionNodes = node.getChildren().stream()
+        .filter(hasType(NodeType.VARIABLE_DEFINITION))
+        .map(VariableDefinitionNode.class::cast)
+        .collect(Collectors.toCollection(LinkedList::new));
+    variableDefinitionNodes.forEach(node::removeChild);
+    List<SyntaxError> errors = new ArrayList<>();
+    errors.addAll(processDefinition(node, 1, variableDefinitionNodes));
+    errors.addAll(checkGlobalUniqueNames(node));
+    errors.addAll(checkTopNumbers(node));
+    return errors;
+  }
+
+  private List<SyntaxError> processDefinition(Node rootNode, int level, Deque<VariableDefinitionNode> definitionNodes) {
```

Comment:
The mnemonic complexity is quite high here, I'm afraid of the new Coverity issue about it.